app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -163,6 +163,9 @@ define(function (require, exports, module) {
     },
 
     deleteDisplayedAccountProfileImage (account) {
+      if ( ! this._displayedProfileImage ) {
```

Comment:
It seems to me this solution sidesteps the problem but does not fix the cause of the problem, which seems to be inconsistent state. `deleteDisplayedAccountProfileImage` is called from [`removeAvatar` in avatar_change.js](https://github.com/mozilla/fxa-content-server/blob/6c6241770b93a8eb6b3cbb9af6f6f3be6c9e7343/app/scripts/views/settings/avatar_change.js#L78). `removeAvatar` can only be called [if we believe the account has a profile image](https://github.com/mozilla/fxa-content-server/blob/6c6241770b93a8eb6b3cbb9af6f6f3be6c9e7343/app/scripts/templates/settings/avatar_change.mustache#L15). `hasProfileImage` is set [if the account has a `profileImageUrl`](https://github.com/mozilla/fxa-content-server/blob/6c6241770b93a8eb6b3cbb9af6f6f3be6c9e7343/app/scripts/views/settings/avatar_change.js#L52). 

So, we believe the account has a `profileImageUrl`, but `this._displayedProfileImage` is undefined, which means [the assignment to `this._displayedProfileImage`](https://github.com/mozilla/fxa-content-server/blob/6c6241770b93a8eb6b3cbb9af6f6f3be6c9e7343/app/scripts/views/mixins/avatar-mixin.js#L73) has not been made.

This can happen because `displayAccountProfileImage` is called in [avatar_change.js->afterVisible](https://github.com/mozilla/fxa-content-server/blob/6c6241770b93a8eb6b3cbb9af6f6f3be6c9e7343/app/scripts/views/settings/avatar_change.js#L58). `afterVisible` is called after the view is already visible (with the Clear button shown to the user), this means there is a window in which the user can click the `Clear` button before the profile image is fetched, and before `this._displayedProfileImage` is set.

This behavior was introduced in #3850 to fix #3543.

We are using the cached data in `this._displayedProfileImage` to get the profile image ID. [The image ID *should* already be available on the account model](https://github.com/mozilla/fxa-content-server/blob/6c6241770b93a8eb6b3cbb9af6f6f3be6c9e7343/app/scripts/models/account.js#L37), which is passed in to this function.

I *believe* the correct fix here is to *not* log and display the error you added here, but rather:

1. Use the image ID from the account model. If there is no image ID on the account model, THEN log and display an error.

OR 

2. Ensure `this._displayedProfileImage` is set by calling `this.displayAccountProfileImage(account)` and then deleting the avatar whenever that function call completes.

We might have to try both approaches to see which works best.